src/components/HeaderLogo/index.tsx

```diff
@@ -0,0 +1,151 @@
+import * as C from './styles';
+import { useState, useEffect, FormEvent } from 'react';
+import { useInfoContext } from '../../contexts/userInfoContext';
+import { checkAccess } from '../../helpers/authorizations';
+import { getPhoto, getTitles, setTitleDatabase, updateTitleDatabase, uploadingPhoto } from '../../database/firebaseLogo';
+import { Photo, TitleLogo, TitleLogoDatabase } from '../../types/logo';
+import { IoMdClose } from 'react-icons/io';
+import Logo from '../../img/logo.png';
+
+type Props = {};
+
+function Headerlogo (props: Props) {
+  // const {} = props;
+  const { state } = useInfoContext();
+
+  const [textLeft, setTextLeft] = useState('Sua');
+  const [textRight, setTextRight] = useState('Marca');
+  const [idTexts, setIdTexts] = useState('');
+  const [logo, setLogo] = useState(Logo);
+  const [uploading, setUploading] = useState(false);
+  const [showFieldSendPhoto, setShowFieldSendPhoto] = useState(false);
+
+  useEffect(() => {
+    if (state.infoUser?.email) {
+      getTitlesDatabase();
+      getLogo();
+    }
+  }, []);
+
+  const getLogo = async () => {
+    const user = state.infoUser?.email;
+    const token = await state.infoUser?.getIdToken();
+    const authorizedDatabase = state.databaseAuth;
+
+    if (authorizedDatabase) {
+      const accessAuthorized = await checkAccess(state);
+      if (!accessAuthorized) {
+        return;
+      }
+    }
```

Comment:
Essas linhas aparecem em alguns endpoints. Acho que vale a pena mover para um middleware. 

```suggestion
    if (authorizedDatabase) {
      const accessAuthorized = await checkAccess(state);
      if (!accessAuthorized) {
        return;
      }
    }
```